libs/common/include/s25util/enumUtils.h

```diff
@@ -4,18 +4,78 @@
 
 #pragma once
 
+#include <boost/config.hpp>
 #include <type_traits>
 
+template<typename Enum>
+struct IsBitset : std::false_type
+{};
+
+template<typename Enum>
+constexpr bool IsValidBitset = IsBitset<Enum>::value && std::is_unsigned<std::underlying_type_t<Enum>>::value;
+
+template<typename Enum>
+using require_validBitset = std::enable_if_t<IsValidBitset<Enum>>;
```

Comment:
Sorry last thing: I want to adhere to the STL convention of using "`_it`" suffix for template aliases resolving the `::type` and `_v` for resolving `::value` such that it is clear on use site

```suggestion
constexpr bool IsValidBitset_v = IsBitset<Enum>::value && std::is_unsigned<std::underlying_type_t<Enum>>::value;

template<typename Enum>
using require_validBitset = std::enable_if_t<IsValidBitset_v<Enum>>;
```